src/data_watcher.cpp

```diff
@@ -278,10 +316,61 @@ bool DataWatcher::processCreate(const eventInfo& receivedEventInfo)
             std::ranges::for_each(
                 fs::recursive_directory_iterator(monitoringPath),
                 modifyWatchIfExpected);
-            return true;
+            return std::make_pair(absCreatedPath, DataOps::COPY);
+        }
+    }
+    return std::nullopt;
+}
+
+std::optional<DataOperation>
+    DataWatcher::processDeleteSelf(const eventInfo& receivedEventInfo)
+{
+    // Case 1 : A monitoring file got deleted.
+    // case 2 : A monitoring directory got deleted.
+    fs::path deletedPath =
+        _watchDescriptors.at(std::get<wd>(receivedEventInfo));
+    lg2::debug("Processing IN_DELETE_SELF for {PATH}", "PATH", deletedPath);
+
+    if (_watchDescriptors.size() == 1)
+    {
+        // Since no more watches will be there, add a watch on parent dir to
+        // notify future create events.
+        addToWatchList(getExistingParentPath(deletedPath),
```

Comment:
I think `_watchDescriptors` will be more than one for the directory, right? If yes, not monitoring the parent directory of configured directory if its deleted?